src/graph/graphcontext.c

```diff
@@ -54,24 +54,26 @@ GraphContext* GraphContext_New(RedisModuleCtx *ctx, const char *graphname,
   return gc;
 }
 
-GraphContext* GraphContext_Retrieve(RedisModuleCtx *ctx, const char *graphname) {
-  GraphContext *gc = NULL;
-  RedisModuleString *rs_name = RedisModule_CreateString(ctx, graphname, strlen(graphname));
-  RedisModuleKey *key = RedisModule_OpenKey(ctx, rs_name, REDISMODULE_READ);
-  if (RedisModule_ModuleTypeGetType(key) != GraphContextRedisModuleType) {
-    goto cleanup;
-  }
-  gc = RedisModule_ModuleTypeGetValue(key);
+GraphContext* GraphContext_Retrieve(RedisModuleCtx *ctx, const char *graphname, bool readonly) {
+    RedisModuleKey *key;
+    GraphContext *gc = NULL;
+    RedisModuleString *rs_name = RedisModule_CreateString(ctx, graphname, strlen(graphname));
 
-  // Force GraphBLAS updates and resize matrices to node count by default
-  Graph_SetMatrixPolicy(gc->g, SYNC_AND_MINIMIZE_SPACE);
+    if(readonly) key = RedisModule_OpenKey(ctx, rs_name, REDISMODULE_READ);
+    else key = RedisModule_OpenKey(ctx, rs_name, REDISMODULE_WRITE);
```

Comment:
The flags that build the `mode` argument are bitmasked, so I think in the write case we probably want to specify `REDISMODULE_READ|REDISMODULE_WRITE`.